tests/expected_tests.cc

```diff
@@ -0,0 +1,39 @@
+#include "gtest/gtest.h"
+#include "stout/expected.h"
+
+stout::expected<int, std::string> divide(int divident, int divider) {
+  if (!divider) {
+    return stout::make_unexpected("divide by zero");
+  } else {
+    return divident / divider;
+  }
+}
+
+TEST(Expected, SucceedResult) {
+  const auto result = divide(6, 3);
+
+  EXPECT_TRUE(result.has_value());
```

Comment:
Optional: we could just use a single expect statement:

`EXPECT_EQ(result, stout::expected<int>(2));`

Not only does this make this code ~shorter, but it also avoids a bad access in the second `EXPECT` statement in the case that `has_value()` returns `false`.

This relies on `tl::expected::operator==`: https://github.com/TartanLlama/expected/blob/96d547c03d2feab8db64c53c3744a9b4a7c8f2c5/include/tl/expected.hpp#L2298-L2300 (which also works with asymmetric types: https://github.com/TartanLlama/expected/blob/96d547c03d2feab8db64c53c3744a9b4a7c8f2c5/include/tl/expected.hpp#L2266-L2271 )

Here and below.

Reply:
I love this cleanup! Thanks!